docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc

```diff
@@ -1,33 +1,305 @@
++
+[source,python]
+----
+
+====
+
+A Data Descriptor for this Complex Number example is again very similar to the ones from previous examples, except the Sample Type might be `ComplexFloat32` or `ComplexFloat64` or  instead of `Float64` or `RangeInt64`.
+
+[#last_value_struct]
+== Getting the last value with a Struct
+
+A Data Descriptor for a Struct might be:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Create a Data Descriptor
+auto descriptor = DataDescriptorBuilder()
+					  .setName("MyStruct")
+					  .setSampleType(SampleType::Struct)
+					  .setStructFields(List<DataDescriptorPtr>(
+						  DataDescriptorBuilder()
+							  .setName("MyInt32")
+							  .setSampleType(SampleType::Int32)
+							  .build(),
+						  DataDescriptorBuilder()
+							  .setName("MyFloat64")
+							  .setSampleType(SampleType::Float64)
+							  .build()))
+					  .build();
+// Set the Data Descriptor, thereby adding MyStruct to the Type Manager
+signal.setDescriptor(descriptor);
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+In the above example, a Data Descriptor of a custom Struct that has two fields is first built. The first field has Sample Type `Int32` and the second one has Sample Type `Float64`.
+
+[CAUTION]
+====
+`opendaq.SampleType.Struct` is not yet supported in Python.
+====
+
+[NOTE]
+====
+Because our Struct represents a custom type, we must call `setDescriptor` (`descriptor` in Python) on our Signal in order to add `MyStruct` to the Type Manager.
+====
+
+[CAUTION]
+====
+All custom Structs and their respective fields must have their names set, which is a deviation from the previous examples. In the above code this is done by the calls to `setName` (`name` in Python) method of the Data Descriptor Builder.
+====
+
+After some data has been sent to our Signal, we can capture a single value similar to the previous examples:
 
 [tabs]
 ====
 Cpp::
 +
 [source,cpp]
 ----
-void HowToDoc_MeasureSingleValue(daq::DevicePtr device)
-{
-    // get 1st signal of device...
-    auto signal = device.getSignalsRecursive()[0];
-    // ...and get a measurement value
-    auto measurementValue = signal.getLastValue();
-    std::cout << "Single measurement value: [" << measurementValue << "]" << std::endl;
-}
+// Get last value of a Signal
+auto lastValue = signal.getLastValue();
+// Cast to IStruct
+auto myStruct = lastValue.asPtr<IStruct>();
+// Extract values
+auto myInt = myStruct.get("MyInt32");
+auto myFloat = myStruct.get("MyFloat64");
 ----
 Python::
 +
 [source,python]
 ----
-def HowToDoc_MeasureSingleValue(device):
-    signal = device.signals_recursive[0]
-    measurementValue = signal.last_value
-    print ("Single measurement value: [",measurementValue,"]")
+
 ----
 ====
 
+[CAUTION]
+====
+If we want to call `getLastValue` (`last_value` in Python) on a Data Packet (as opposed to a Signal) and the underlying data type is a Struct, we must provide the optional parameter, Type Manager, which has had the custom Struct added to it.
```

Comment:
Honestly, I would not even mention that we can read the last value on a packet. The guide aims to describe how to read the last value of a signal, the packet last value is mostly for internal usage and will confuse readers.